Source/FieldSolver/WarpXPushFieldsHybridPIC.cpp

```diff
@@ -189,6 +189,18 @@ void WarpX::HybridPICEvolveFields ()
                            0, 0, 1, current_fp_temp[lev][idim]->nGrowVect());
         }
     }
+
+    // Check that the E-field does not have nan or inf values, otherwise print a clear message
+    ablastr::fields::MultiLevelVectorField Efield_fp = m_fields.get_mr_levels_alldirs(FieldType::Efield_fp, finest_level);
+    for (int lev = 0; lev <= finest_level; ++lev)
+    {
+        for (int idim = 0; idim < 3; ++idim) {
+            WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
+                Efield_fp[lev][idim]->is_finite(),
+                "Non-finite value detected in E-field; this indicates more substeps should be used in the field solver."
```

Comment:
We could:
- Set the flag `local=true` to avoid parallel operations (reduction)
- Use `amrex::Abort` so that it does not hand if only one of the MPI ranks fails.

Reply:
I tested that approach and it still hangs if all processors don't call `Abort`.